packages/core/src/utils/decoder.ts

```diff
@@ -2,26 +2,35 @@ import '@polkadot/types-codec'
 import { Block } from '../blockchain/block.js'
 import { DecoratedMeta } from '@polkadot/types/metadata/decorate/types'
 import { HexString } from '@polkadot/util/types'
+import { LRUCache } from 'lru-cache'
 import { StorageEntry } from '@polkadot/types/primitive/types'
 import { StorageKey } from '@polkadot/types'
 import { hexToU8a, u8aToHex } from '@polkadot/util'
 import _ from 'lodash'
 
 import { decodeWellKnownKey } from './well-known-keys.js'
 
-const _CACHE: Record<string, Map<HexString, StorageEntry>> = {}
+const _CACHE: Record<string, LRUCache<HexString, StorageEntry>> = {}
 
-const getCache = (uid: string): Map<HexString, StorageEntry> => {
+function createCache() {
+  return new LRUCache<HexString, StorageEntry>({
+    max: 500, // The maximum number of items in the cache
```

Comment:
can we do some quick tests to see if this number is good?
basically make max to a very big number, run bunch blocks, check cache size and memory usage

Reply:
I tested with some polkadot and acala blocks, the number of items in this cache rarely go up to more than 50, so I changed max to 50. but memory usage keeps growing, there are other places that is eating the memory (assuming lru-cache is working fine)